cachi2/core/package_managers/gomod.py

```diff
@@ -0,0 +1,1022 @@
+
+    invalid_gomod_files = _find_missing_gomod_files(request.source_dir, subpaths)
+
+    if invalid_gomod_files:
+        invalid_files_print = "; ".join(invalid_gomod_files)
+        file_suffix = "s" if len(invalid_gomod_files) > 1 else ""
+
+        # missing gomod files is supported if there is only one path referenced
+        if config.cachito_gomod_ignore_missing_gomod_file and len(subpaths) == 1:
+            log.warning("go.mod file missing for request at %s", invalid_files_print)
+            return
+
+        raise FileAccessError(
+            "The {} file{} must be present for the gomod package manager".format(
+                invalid_files_print.strip(), file_suffix
+            )
+        )
+
+    if len(subpaths) > 1 and request.dep_replacements:
+        raise UnsupportedFeature(
+            "Dependency replacements are only supported for a single go module path."
+        )
+
+    env_vars = {
+    "GOCACHE": {"value": "deps/gomod", "kind": "path"},
+    "GOPATH": {"value": "deps/gomod", "kind": "path"},
+    "GOMODCACHE": {"value": "deps/gomod/pkg/mod", "kind": "path"},
+    }
+    env_vars.update(config.cachito_default_environment_variables.get("gomod", {}))
+
+    packages_json_data = PackagesData()
+
+    for i, subpath in enumerate(subpaths):
+        log.info(
+            "Fetching the gomod dependencies for request in subpath %s", subpath
+        )
+
+        log.info(f'Fetching the gomod dependencies at the "{subpath}" directory')
+
+        gomod_source_path = request.source_dir / subpath
+        try:
+            gomod = _resolve_gomod(
+                gomod_source_path, request
+            )
+        except GoModError:
+            log.exception("Failed to fetch gomod dependencies for request %d")
+            raise
+
+        module_info = gomod["module"]
+
+        packages_json_data.add_package(module_info, subpath, gomod["module_deps"])
+
+        # add package deps
+        for package in gomod["packages"]:
+            pkg_info = package["pkg"]
+            package_subpath = _package_subpath(module_info["name"], pkg_info["name"], subpath)
+            packages_json_data.add_package(pkg_info, package_subpath, package.get("pkg_deps", []))
+
+    packages_json_data.write_to_file(request.output_dir / "gomod_packages.json")
+
+
+
+def _find_missing_gomod_files(source_path: Path, subpaths: list[str]):
+    """
+    Find all go modules with missing gomod files.
+
+    These files will need to be present in order for the package manager to proceed with
+    fetching the package sources.
+
+    :param RequestBundleDir bundle_dir: the ``RequestBundleDir`` object for the request
+    :param list subpaths: a list of subpaths in the source repository of gomod packages
+    :return: a list containing all non-existing go.mod files across subpaths
+    :rtype: list
+    """
+    invalid_gomod_files = []
+    for subpath in subpaths:
+        package_gomod_rel_path = source_path / subpath / "go.mod"
```

Comment:
nitpick: this "relpath" is not actually [guaranteed to be] relative

Reply:
Fixed it.